src/Symfony/Component/Translation/MessageCatalogueProvider/CacheMessageCatalogueProvider.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\MessageCatalogueProvider;
+
+use Symfony\Component\Config\ConfigCacheInterface;
+use Symfony\Component\Config\ConfigCacheFactoryInterface;
+use Symfony\Component\Config\ConfigCacheFactory;
+use Symfony\Component\Translation\MessageCatalogue;
+
+/**
+ * Manages cache catalogues.
+ *
+ * @author Abdellatif Ait boudad <a.aitboudad@gmail.com>
+ */
+class CacheMessageCatalogueProvider implements MessageCatalogueProviderInterface
+{
+    /**
+     * @var MessageCatalogueProviderInterface
+     */
+    private $messageCatalogueProvider;
+
+    /**
+     * @var string
+     */
+    private $cacheDir;
+
+    /**
+     * @var ConfigCacheFactoryInterface
+     */
+    private $configCacheFactory;
+
+    /**
+     * @param MessageCatalogueProviderInterface $messageCatalogueProvider The catalogue bag to use for loading the catalogue.
+     * @param string                            $cacheDir                 The directory to use for the cache.
+     * @param bool                              $debug                    Use cache in debug mode.
```

Comment:
Take a ConfigCacheFactoryInstance in the constructor and let your clients decide. This class is a decorator that adds caching to a MessagCatalogueProvider and IMO it's perfectly fine if the ConfigCacheFactory is a required argument.